scrapy/contracts/__init__.py

```diff
@@ -84,8 +86,11 @@ def cb_wrapper(response):
 
         def eb_wrapper(failure):
             case = _create_testcase(method, 'errback')
-            exc_info = failure.value, failure.type, failure.getTracebackObject()
-            results.addError(case, exc_info)
+            exc_info = failure.type, failure.value, failure.getTracebackObject()
+            if failure.type.__name__ in ignored_errors:
```

Comment:
A nitpick: what about making `ignored_errors` a set instead of a list (in from_method and IgnoreContract constructor)? Length shouldn't be large, so it doesn't matter much, but set looks more clear.